types/iyzipay/index.d.ts

```diff
@@ -0,0 +1,1601 @@
+    ): void;
+    initialize(
+        data: ISubscriptionInitializeRequestData,
+        callback: (err: Error, result: ISubscriptionCheckoutFormInitializeResult) => void,
+    ): void;
+}
+
+export interface ISubscriptionCheckoutForm {
+    initialize(
+        data: ISubscriptionCheckoutFormInitializeRequestData,
+        callback: (err: Error, result: ISubscriptionCheckoutFormInitializeResult) => void,
+    ): void;
+    retrieve(
+        data: ISubscriptionCheckoutFormRetrieveRequestData,
+        callback: (err: Error, result: ISubscriptionCheckoutFormRetrieveResult) => void,
+    ): void;
+}
+
+export interface ISubscriptionExistingCustomer {
+    initialize(
+        data: ISubscriptionExistingCustomerInitializeRequestData,
+        callback: (err: Error, result: ISubscriptionExistingCustomerInitializeResult) => void,
+    ): void;
+}
+
+export class Iyzipay {
+    constructor(options?: IConfigOptions);
+    apiTest: IApiTest;
+    approval: IApproval;
+    disapproval: IDisapproval;
+    bkmInitialize: IBKMInitialize;
+    bkm: IBKM;
+    cancel: ICancel;
+    card: ICard;
+    cardList: ICardList;
+    checkoutFormInitialize: ICheckOutFormInitialize;
+    checkoutForm: ICheckOutForm;
+    universalCardStorageInitialize: IUniversalCardStorageInitialize;
+    installmentInfo: IInstallmentInfo;
+    payment: IPayment;
+    paymentItem: IPaymentItem;
+    peccoInitialize: IPeccoInitialize;
+    peccoPayment: IPeccoPayment;
+    refund: IRefund;
+    refundToBalance: IRefundToBalance;
+    payoutCompletedTransactionListRetrieve: IPayoutCompletedTransactionListRetrieve;
+    bouncedBackTransactionListRetrieve: IBouncedBackTransactionListRetrieve;
+    subMerchant: ISubMerchant;
+    threedsInitialize: IThreeDSInitialize;
+    threedsPayment: IThreedsPayment;
+
+    settlementToBalance: ISettlementToBalance;
+    /* SUBSCRIPTION */
+    subscriptionProduct: ISubscriptionProduct;
+    subscriptionPricingPlan: ISubscriptionPricingPlan;
+    subscriptionCustomer: ISubscriptionCustomer;
+    subscriptionCard: ISubscriptionCard;
+    subscriptionPayment: ISubscriptionPayment;
+    subscription: ISubscription;
+    subscriptionCheckoutForm: ISubscriptionCheckoutForm;
+    subscriptionExistingCustomer: ISubscriptionExistingCustomer;
+}
+
+export namespace Iyzipay {
+    const LOCALE: ILOCALE;
+    const PAYMENT_GROUP: IPAYMENT_GROUP;
+    const BASKET_ITEM_TYPE: IBASKET_ITEM_TYPE;
+    const PAYMENT_CHANNEL: IPAYMENT_CHANNEL;
+    const SUB_MERCHANT_TYPE: ISUB_MERCHANT_TYPE;
+    const CURRENCY: ICURRENCY;
+    const APM_TYPE: IAPM_TYPE;
+    const REFUND_REASON: IREFUND_REASON;
+    const PLAN_PAYMENT_TYPE: IPLAN_PAYMENT_TYPE;
+    const SUBSCRIPTION_PRICING_PLAN_INTERVAL: ISUBSCRIPTION_PRICING_PLAN_INTERVAL;
+    const SUBSCRIPTION_UPGRADE_PERIOD: ISUBSCRIPTION_UPGRADE_PERIOD;
+    const SUBSCRIPTION_STATUS: ISUBSCRIPTION_STATUS;
+    const SUBSCRIPTION_INITIAL_STATUS: ISUBSCRIPTION_INITIAL_STATUS;
+}
+
+export default Iyzipay;
```

Comment:
this is CJS module, it seems:
https://github.com/iyzico/iyzipay-node/blob/master/lib/Iyzipay.js#L133
So I'd consider modelling along TS handbook guideline (or CJS module template for decoration types):
https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html

something along this pseudo-code:
```ts
declare class Iyzipay {
...
}

declare namespace Iyzipay {
...
}

export = Iyzipay;
```

Reply:
I have removed commonjs, instead I used es6. It seems I looks like I dont need commonjs. I commited the latest changes